push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/VariantMetricCollector.java

```diff
@@ -0,0 +1,47 @@
+package org.jboss.aerogear.unifiedpush.message;
+
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+
+import org.jboss.aerogear.unifiedpush.api.PushMessageInformation;
+import org.jboss.aerogear.unifiedpush.api.VariantMetricInformation;
+import org.jboss.aerogear.unifiedpush.message.jms.Dequeue;
+import org.jboss.aerogear.unifiedpush.service.metrics.PushMessageMetricsService;
+
+public class VariantMetricCollector {
```

Comment:
I'll add javadoc,

`Sender` MDB does not directly write variant metrics into database, as it could simply cause contension and failing transactions. Instead, it fires message that acts as an request for counter update in DB.

See https://docs.google.com/a/fryc.eu/drawings/d/13IsJWPSJNYXtst-UVxQYmzH36C_EXQMYYr_jcu7nFmE/edit